src/PhpParser/Node/BetterNodeFinder.php

```diff
@@ -346,18 +339,36 @@ public function findFirstPreviousOfTypes(Node $mainNode, array $types): ?Node
         );
     }
 
+    private function resolveNextNode(Node $node, ?Node $parentNode): ?Node
+    {
+        if (! $parentNode instanceof Node) {
+            $newStmts = $this->resolveNewStmts($parentNode);
+            return $this->resolveNodeFromFile($newStmts, $node, false);
+        }
+
+        if ($node instanceof Stmt) {
+            if (! $parentNode instanceof StmtsAwareInterface) {
+                return null;
+            }
+
+            // todo: update to use key +1
+            return $node->getAttribute(AttributeKey::NEXT_NODE);
```

Comment:
one more to go 👍 , it seems tricky as the node overlapped to just removed, changing to:

```php
            $currentStmtKey = $node->getAttribute(AttributeKey::STMT_KEY);
            return $parentNode->stmts[$currentStmtKey+1] ?? null;
```

got error in test:

```diff
There was 1 failure:

1) Rector\Tests\DeadCode\Rector\Assign\RemoveUnusedVariableAssignRector\RemoveUnusedVariableAssignRectorTest::test with data set #13
Failed on fixture file "unused_local_var.php.inc"
Failed asserting that string matches format description.
--- Expected
+++ Actual
@@ @@
 {
     public function run()
     {
-        $test = date('Y-m-d');
         echo $test
```

Reply:
Fixed by [check start token pos](https://github.com/rectorphp/rector-src/pull/3879/commits/cdfb75c660ac30f8278f97dc7293019296be0489)